### PR TITLE
Transition to standard smart pointers, part 9

### PR DESCRIPTION
**Pull request by taketwo**
Partially addresses #2792.

### CODE REVIEW

**Comment by SergioRAgostinho** [comment]
I'll start reviewing once the CI error is cleared.
```
[ 31%] Building CXX object tools/CMakeFiles/pcl_mesh_sampling.dir/mesh_sampling.cpp.o
/__w/1/s/tools/openni2_viewer.cpp: In function 'int main(int, char**)':
/__w/1/s/tools/openni2_viewer.cpp:310:48: error: 'pcl::io::openni2::OpenNI2DeviceManager' has not been declared
         auto deviceManager = pcl::io::openni2::OpenNI2DeviceManager::getInstance ();
                                                ^
/__w/1/s/tools/openni2_viewer.cpp:330:44: error: 'pcl::io::openni2::OpenNI2DeviceManager' has not been declared
     auto deviceManager = pcl::io::openni2::OpenNI2DeviceManager::getInstance ();
```

**Comment by taketwo** [comment]
Done!

File: io/src/ply/ply_parser.cpp [hunk 1]
```diff
@@ -182,14 +182,12 @@ bool pcl::io::ply::ply_parser::parse (const std::string& filename)
...
         }
-        std::vector< boost::shared_ptr<element> >::const_iterator iterator;
-        for (iterator = elements.begin (); iterator != elements.end (); ++iterator)
+        auto iterator = elements.cbegin ();
+        while (iterator != elements.cend ())
         {
-          const struct element& element = *(iterator->get ());
-          if (element.name == name)
-          {
+          if (iterator->get ()->name == name)
```
Comment (SergioRAgostinho) [inline io/src/ply/ply_parser.cpp]:
```suggestion
          if ((*iterator)->name == name)
```

File: io/src/ply/ply_parser.cpp [hunk 2]
```diff
@@ -248,14 +242,12 @@ bool pcl::io::ply::ply_parser::parse (const std::string& filename)
...
-          std::vector< boost::shared_ptr<property> >::const_iterator iterator;
-          for (iterator = current_element_->properties.begin (); 
-               iterator != current_element_->properties.end (); 
-               ++iterator)
+          auto iterator = current_element_->properties.cbegin ();
+          while (iterator != current_element_->properties.cend ())
           {
-            const struct property& property = *(iterator->get ());
-            if (property.name == name)
+            if (iterator->get ()->name == name)
```
Comment (SergioRAgostinho) [inline io/src/ply/ply_parser.cpp]:
```suggestion
            if ((*iterator)->name == name)
```

File: io/src/ply/ply_parser.cpp [hunk 3]
```diff
@@ -326,14 +318,12 @@ bool pcl::io::ply::ply_parser::parse (const std::string& filename)
...
-          std::vector< boost::shared_ptr<property> >::const_iterator iterator;
-          for (iterator = current_element_->properties.begin (); 
-               iterator != current_element_->properties.end (); 
-               ++iterator) 
+          auto iterator = current_element_->properties.cbegin ();
+          while (iterator != current_element_->properties.cend ())
           {
-            const struct property& property = *(iterator->get ());
-            if (property.name == name)
+            if (iterator->get ()->name == name)
```
Comment (SergioRAgostinho) [inline io/src/ply/ply_parser.cpp]:
```suggestion
            if ((*iterator)->name == name)
```

File: io/include/pcl/io/openni2/openni2_device_manager.h [hunk 4]
```diff
@@ -56,16 +57,16 @@ namespace pcl
 
         // This may not actually be a singleton yet. Need to work out cross-dll incerface.
         // Based on http://stackoverflow.com/a/13431981/1789618
-        static boost::shared_ptr<OpenNI2DeviceManager> getInstance ()
+        static std::shared_ptr<OpenNI2DeviceManager> getInstance ()
         {
-          static boost::shared_ptr<OpenNI2DeviceManager> instance = boost::make_shared<OpenNI2DeviceManager>();
+          static auto instance = std::make_shared<OpenNI2DeviceManager>();
```
Comment (SergioRAgostinho) [inline io/include/pcl/io/openni2/openni2_device_manager.h]: Self: Revise later if replaced by `pcl::make_shared`.

File: io/include/pcl/io/davidsdk_grabber.h [hunk 5]
```diff
@@ -75,13 +75,13 @@ namespace pcl
 
       using sig_cb_davidsdk_mesh = void() (const pcl::PolygonMesh::Ptr &);
 
-      using sig_cb_davidsdk_image = void() (const boost::shared_ptr<pcl::PCLImage> &);
+      using sig_cb_davidsdk_image = void() (const pcl::PCLImage::Ptr &);
```
[the remark this answers is not in the document]
Comment (taketwo) [reply io/include/pcl/io/davidsdk_grabber.h]: Good point, see #3225.

File: io/include/pcl/io/openni2/openni2_device_manager.h (hunk 4 above)
Comment (taketwo) [reply io/include/pcl/io/openni2/openni2_device_manager.h]: I'll switch to `pcl::make_shared`. I think our policy should be "always use pcl::make_shared, no matter if there is a custom allocator or not" .

**Comment by taketwo** [comment]
Rebased and solved the conflicts.

**Pull request merged**